nvme.c

```diff
@@ -328,6 +333,7 @@ static int get_telemetry_log(int argc, char **argv, struct command *cmd, struct
 	 */
 	if (full_size % bs) {
 		fprintf(stderr, "The full size of the data area requested is not rounded to the nearest block\n");
+		free(page_log);
```

Comment:
This one is OK,  but moving free(page_log), into free_hdr might seem better as well as renaming free_hdr into free_mem. :) never mind, it's just personal preference.

Reply:
I agree with you. I was going to do that, but noticed that on line 289 we use goto free_hdr, and there is no page_log allocated yet. I was worried that attempting to free memory that has not been allocated would cause problems.